### DISCUSSION TITLE
how to self host it on vercel?

### DISCUSSION THREAD

**Discussion by Arnav717** [alfaarghya/alfa-leetcode-api#21] [Q&A]
I am having trouble self hosting it 
can anyone help me with this?

**Comment by alfaarghya** [comment]
@Arnavop check out the **V1.0.0 branch**, you will find a vercel.json file Simply add that to your code.

‼️ if it still gives you an error try to go through package.json of the **V1.0.0 branch**.

**Comment by Arnav717** [comment]
idk it gives me error
404: NOT_FOUND

**Reply by alfaarghya** [reply]
Then I don't know 😥

**Reply by merakesh99** [reply]
> Then I don't know 😥

Could you please explain the issue?

**Reply by Arnav717** [reply]
> > Then I don't know 😥
> 
> Could you please explain the issue?

I am not able to host on vercel

**Reply by alfaarghya** [reply]
Hi @Arnavop can u tell me when it gives u 404 error?

**Reply by Arnav717** [reply]
After i host
Like after i run it and open the website after it gets deployed

**Reply by alfaarghya** [reply]
Are you trying to host this repo or your own website?

**Reply by Arnav717** [reply]
This repo

**Reply by alfaarghya** [reply]
@Arnavop Ok then try to do one thing.

Goto `package.json`  -> the change `main: src/index.ts`

Then try to deploy it. . . . .  And let me!

**Comment by alfaarghya** [comment]
> I am having trouble self hosting it 
> can anyone help me with this?

@Arnavop did u connect your GitHub with vercel?

**Reply by Arnav717** [reply]
> Ye bro or else how would i able to host it

**Reply by alfaarghya** [reply]
Or By drag and drop the entire code on Vercel 

I am asking because I tried this method and I got an error for the first time

**Reply by Arnav717** [reply]
Well I connected GitHub to vercel and then hosted the forked repo

**Comment by yeora26** [comment]
Hello, I've encountered the same error
<img width="1652" height="1300" alt="image" src="https://github.com/user-attachments/assets/b2cd72a7-ac80-4170-a0b4-fc033a8aac14" />

<img width="2122" height="1434" alt="image" src="https://github.com/user-attachments/assets/99e404ab-59f5-4afc-a354-c23a4a6a4224" />

**Reply by DarrylMathias** [reply]
Usually I have seen, vercel's serverless hosting create's issues hosting server backend's like ones made with Express. I guess it's the same issue here. Have you tried hosting it on Render or Railway?
